build_tools/cmake/iree_benchmark_suite.cmake

```diff
@@ -284,23 +291,37 @@ function(iree_benchmark_suite)
         COMMENT "Generating ${_TOOL_FILE}"
       )
 
-      set(_TOOLFILE_GEN_TARGET_NAME_LIST "iree-generate-benchmark-toolfile")
-      list(APPEND _TOOLFILE_GEN_TARGET_NAME_LIST ${_COMMON_NAME_SEGMENTS})
-      list(JOIN _TOOLFILE_GEN_TARGET_NAME_LIST "__" _TOOLFILE_GEN_TARGET_NAME)
+      set(_TOOLFILE_GEN_TARGET_NAME
+        "${PACKAGE_NAME}_iree-generate-benchmark-toolfile__${_RUN_SPEC_TARGET_SUFFIX}")
       add_custom_target("${_TOOLFILE_GEN_TARGET_NAME}"
         DEPENDS "${_TOOL_FILE}"
       )
 
       # Generate a flagfile containing command-line options used to compile the
       # generated artifacts.
-      set(_COMPOPT_FILE "${_RUN_SPEC_DIR}/compilation_flagfile")
-      string(REPLACE ";" "\n" IREE_BENCHMARK_COMPILATION_FLAGS "${_TRANSLATION_ARGS}")
-      configure_file(
-        ${PROJECT_SOURCE_DIR}/build_tools/cmake/benchmark_compilation_flagfile.in
-        ${_COMPOPT_FILE})
+      string(REPLACE ";" "\\n" _BENCHMARK_COMPILATION_FLAGS "${_TRANSLATION_ARGS}")
+      set(_COMPILATION_FLAGFILE "${_RUN_SPEC_DIR}/compilation_flagfile")
+      add_custom_command(
+        OUTPUT "${_COMPILATION_FLAGFILE}"
+        COMMAND ${CMAKE_COMMAND} -E echo -e -- ${_BENCHMARK_COMPILATION_FLAGS} > "${_COMPILATION_FLAGFILE}"
```

Comment:
L302 transforms the list of flags into "<flag1>\\n<flag2>\\n...", `echo -e` will then turn `\n` into the real newlines.
This should be fine if each flag is only one line (no newline character in the flag).

The reason we can't use `configure_file` or `file` here is because they will always generate the files regardless of the targets. So it will create the incomplete benchmark suite directories if we only compile for one architecture.

Reply:
Eventually add a small python script for this. Let me know if you prefer other ways.

Reply:
Well I'd really prefer that CMake allowed you to easily do simple things, but alas that will never be